### ISSUE TITLE
Operations fail on /Applications due to lack of permissions.

### ISSUE THREAD

**Issue by ivansavra1** [alienator88/Sentinel#9]
Hi, thanks for making Sentinel, it's a pretty useful app, I wanted to suggest you to look into error handling and sudo permissions. When you try to act on a file that's on /Applications (And other system directories), since Sentinel lacks permission to modify files there, the operations fail but the "App was removed from quarantine / signed" message still shows, it seems there is no error handling, and it automatically assumes it was successful after running the command or call.

I know this is the case, because I run `xattr "/Applications/something.app"` and I can see that the quarantine attribute is still there, so in addition to the error handling, there should be a fallback that uses sudo (Getting the correct permission using Touch ID / password) so it works in more cases.

Thanks.

**Comment by alienator88** [comment]
Yeah, the there's no error handling there, I'm not sure I even used that function much myself. This was my first Mac app so there's probably some things missing 😂 I'll take a look tomorrow.
If you launch the app from Terminal and reproduce the issue, it might show the console log error of exactly what's happening.

**Comment by ivansavra1** [comment]
> This was my first Mac app
You did a good job considering that!

**Comment by alienator88** [comment]
@ivansavra1 would you be able to test this build out for me?
I'm unable to reproduce your issue with apps in /Applications so it's a best guess effort currently.
[Sentinel.zip](https://github.com/user-attachments/files/17907340/Sentinel.zip)

If it still doesn't work, I added some debug logging. Launch the app from Terminal by running: `/Applications/Sentinel/Contents/MacOS/Sentinel`

This should show logging in the terminal window when it launches. Then drop one of the apps that don't work to remove from quarantine and see what errors it spits out.

**Comment by alienator88** [comment]
No reply, closing this out.
New build is released.
